### PR TITLE
Update error codes

### PR DESCRIPTION
**Pull request by calummoore**

### CODE REVIEW

File: packages/client/src/__tests__/Polybase.spec.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ test('creates collections from schema in namespace', async () => {
       status: 400,
       data: {
         error: {
-          reason: 'record-not-found',
+          reason: 'document/not-found',
           code: 'not-found',
           message: 'Collection@Col not found: collection record not found: key not found: pebble: not found',
```
Comment (mateuszmlc) [inline packages/client/src/__tests__/Polybase.spec.ts:68]: Here we still call the entries/items of collections "records". Do we use both or always call them documents from now on?

**Review by mateuszmlc — APPROVED** [review]

File: packages/client/src/__tests__/Polybase.spec.ts (hunk 1 above)
Comment (calummoore) [reply packages/client/src/__tests__/Polybase.spec.ts:68]: We should always call them documents from now on to be consistent

File: packages/client/src/Collection.ts [hunk 2]
```diff
@@ -84,15 +84,20 @@ export class Collection<T> {
...
     return res.data
   }
 
-  doc = (id: string): Doc<T> => {
-    return new Doc<T>(id, this, this.client, this.onDocSnapshotRegister)
+  record = (id: string): CollectionRecord<T> => {
+    return new CollectionRecord<T>(id, this, this.client, this.onCollectionRecordSnapshotRegister)
+  }
+
+  // Deprecated
```
Comment (mateuszmlc) [inline packages/client/src/Collection.ts:98]: We could use the @deprecated jsdoc annotation so editors like VS Code show this function with a strikethrough if you use it.
https://jsdoc.app/tags-deprecated.html

File: packages/client/src/Record.ts [hunk 3]
```diff
@@ -50,21 +50,23 @@ export class Doc<T> {
...
   }
 
   request = (): Request => ({
-    url: `/collections/${encodeURIComponent(this.collection.id)}/documents/${encodeURIComponent(this.id)}`,
+    url: `/collections/${encodeURIComponent(this.collection.id)}/records/${encodeURIComponent(this.id)}`,
     method: 'GET',
   })
 }
+
+export const Doc = CollectionRecord
```
Comment (mateuszmlc) [inline packages/client/src/Record.ts:72]:
```suggestion
/**
 * @deprecated
 */
export const Doc = CollectionRecord
```

**Review by mateuszmlc — APPROVED** [review]

**Pull request merged**